script.js

```diff
@@ -1,4 +1,284 @@
+    }
+    if (numberOfDice != '') {
+      populateRandomDice(currentPlayer, currentRound, numberOfDice);
+      orderMode = 3; // 3 is the index for "auto highest"; refer to definition under global variable declaration
+      getCombinedNumber(orderMode, currentPlayer, currentRound);
+    }
+  }
+
+  var message = displayMessage(gameMode, orderMode, currentPlayer);
+
+  // When it reaches end of the current player
+  if (
+    ((gameMode == 1 || gameMode == 2) && (input == 1 || input == 2)) ||
+    (gameMode != 1 && gameMode != 2 && numberOfDice != '')
+  ) {
+    currentPlayer = nextPlayer;
+    orderMode = 0;
+    message =
+      message +
+      `<br> <br> It is now Player ${nextPlayer}'s turn. Please click Submit to play`;
+    // When it is also the end of the round
+    if (nextPlayer == 1) {
+      currentRound++;
+      numberOfDice = '';
+      winnerIndex = getWinner(gameMode);
+      message = message + `<br> <br> ${displayLeaderboard(winnerIndex)} `;
+    }
+    if (nextPlayer != numberOfPlayers) {
+      nextPlayer++;
+    } else {
+      nextPlayer = 1;
+    }
+  }
+
+  return message;
+};
+
+// General output message
+var displayMessage = function (gameMode, order, playerIndex) {
+  var message = '';
+  var latestNumber;
+  if (numberOfPlayers == '') {
+    message = `How many players are playing this game? Please enter integers (e.g. 1, 2, 3 ...)`;
+  } else if (numberOfDice == '') {
+    message = `How many dices do you want to play in round ${currentRound}? Please enter integers (e.g. 1, 2, 3 ...)`;
+  } else if (order == 1 || order == 2) {
+    latestNumber = combinedDiceValue[combinedDiceValue.length - 1].value;
+    message = `Player ${playerIndex}, you chose Dice ${order} first. <br> Your number is ${latestNumber}.`;
+  } else {
+    message = `Welcome Player ${playerIndex}. <br> You rolled ${rollMessage}. <br>`;
+    if (gameMode == 1 || gameMode == 2) {
+      message = message + `Choose the Dice order by entering 1 or 2`;
+    } else {
+      latestNumber = combinedDiceValue[combinedDiceValue.length - 1].value;
+      message =
+        message +
+        `Player ${playerIndex}, your automatically combined number is ${latestNumber}.`;
+    }
+  }
+  return message;
+};
+
+// Tally score and determine winner
+var getWinner = function (gameMode) {
+  runningScore = [];
+  // Tally score for each player
+  var playerIndex = 1;
+  while (playerIndex <= numberOfPlayers) {
+    var tally = 0;
+    var counter = 0;
+    while (counter < combinedDiceValue.length) {
+      if (combinedDiceValue[counter].player == playerIndex) {
+        tally = tally + combinedDiceValue[counter].value;
+      }
+      counter++;
+    }
+    runningScore.push({
+      player: playerIndex,
+      value: tally,
+    });
+    playerIndex++;
+  }
+  // Sort running score array for (1) leaderboard, (2) winner can be conveniently determined from index 0 position
+  var x = 0;
```

Comment:
Really good work coming up with the sorting logic! For such complex logic, it's a good idea to extract it out into a helper function outside the main function so you can give it an intuitive name and makes the code easier to read as well 😄

Reply:
What you just did is called a bubble sort! Mad respect! https://www.geeksforgeeks.org/bubble-sort/